src/main/java/gomedic/commons/core/Messages.java

```diff
@@ -33,29 +35,131 @@
+        Iterator<String> iterator = approvedSuggestions.iterator();
         if (iterator.hasNext()) {
             String additionalReply = " You can choose from these commands instead: \n";
             while (iterator.hasNext()) {
-                additionalReply += iterator.next().getValue() + "\n";
+                additionalReply += iterator.next() + "    ";
             }
             reply += additionalReply;
         }
 
         return reply;
     }
 
+    /**
+     * Returns a list of approved command suggestions based on type.
+     *
+     * @param command First part of erroneous command input.
+     * @param listOfTypes Hashset of command types to refer to.
+     * @param listOfTargets List of command targets to be appended to approved suggestions.
+     * @return A list of approved suggestions.
+     */
+    private static List<String> generateTypeSuggestions(String command, HashSet<String> listOfTypes,
+                                                    List<String> listOfTargets) {
+        List<Pair<Integer, String>> closestStrings;
+        LevenshteinDistance stringChecker = new LevenshteinDistance();
+        HashSet<String> singleWordCommands = new HashSet<>(Arrays.asList("exit", "help"));
+
+        // get a list of pairs of (levenshtein distance, type suggestion) sorted by distance
+        closestStrings = listOfTypes.stream()
+                .map(x -> new Pair<>(stringChecker.apply(x, command), x))
+                .sorted(Comparator.comparingInt(Pair::getKey))
+                .collect(Collectors.toList());
+        // list of valid suggestions only if distance is less than or equal to half the suggested command
+        List<Pair<Integer, String>> validTypes = closestStrings.stream()
+                .filter(x -> x.getKey() <= Math.ceil(x.getValue().length() / 2))
+                .limit(5)
+                .collect(Collectors.toList());
+        // append second part of command to available command types
+        return validTypes.stream()
+                .flatMap(x -> !singleWordCommands.contains(x.getValue())
+                        ? listOfTargets
+                        .stream()
+                        .map(y -> x.getValue() + " " + y)
+                        : Arrays.asList(x.getValue()).stream())
+                .limit(5)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a list of command suggestions based on target.
+     *
+     * @param command Second part of erroneous input.
+     * @param listOfTypes
+     * @param listOfTargets
+     * @return
+     */
+    private static List<String> generateTargetSuggestions(String command, HashSet<String> listOfTypes,
+                                                        List<String> listOfTargets) {
+        List<Pair<Integer, String>> closestStrings;
+        LevenshteinDistance stringChecker = new LevenshteinDistance();
+        HashSet<String> singleWordCommands = new HashSet<>(Arrays.asList("exit", "help"));
```

Comment:
Actually I think a lot of parts are kind of duplicate with generateTypeSuggestions, but for now I think its fine, we can refine this in v1.4 or if we have time

Reply:
Hem, I think is pretty minor we can change it later, thanks man